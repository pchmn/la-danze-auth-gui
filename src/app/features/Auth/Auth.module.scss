.Auth {

  height: 100%;
  padding-top: 40px;

  @media (max-width: 530px) {
    padding-top: 20px;
  }

  .logo {
    display: block;
    margin-right: auto;
    margin-left: auto;
  }

  .signInHelper {
    display: block;
    text-align: center;
    margin-top: 40px;
    margin-bottom: 10px;
    color: #979797;
    font-size: 0.9rem;

    @media (max-width: 530px) {
      margin-top: 20px;
    }

    a {
      font-weight: 500;
      font-size: 0.95rem;
      margin-left: 5px;
    }
  }

  .paper {
        
    @media (max-width: 530px) {
      padding: 50px;
      height: 100%;
      border-bottom-left-radius: 0;
      border-bottom-right-radius: 0;
    }
  
    .title {
      margin-top: 0;
      margin-bottom: 50px;
      font-size: 2rem;
      font-weight: 400;

      @media (max-width: 530px) {
        font-size: 1.8rem;
      }
    }
  
    .form {
      margin: auto;
  
      .textField {
        display: block;
        margin-bottom: 25px;
      }
  
      .submitButton {
        margin-top: 10px;
      }
    }

    .progress {
      position: absolute;
      bottom: 0px;
      left: 0;
      width: 100%;
    }
  }
}