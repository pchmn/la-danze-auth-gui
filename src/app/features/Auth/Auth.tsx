import Container from '@material-ui/core/Container';
import Paper from '@material-ui/core/Paper';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { BrowserRouter, Link, useRouteMatch } from 'react-router-dom';
import { AnimatedRoute, AnimatedSwitch } from '../../shared/animation/RouteTransition';
import styles from './Auth.module.scss';
import { SignIn } from './SignIn/SignIn';
import { SignUp } from './SignUp/SignUp';

export function Auth() {
  const { t } = useTranslation();
  const matches = useMediaQuery('(max-width: 530px)');
  let { url } = useRouteMatch();
  console.log('in Auth', url)

  return (
    <>
      <Container className={`${styles.Auth} ${matches ? 'flex-column' : ''}`} maxWidth="sm">
        <img className={styles.logo} alt="logo" src="/images/logo-48px.png" />
        <BrowserRouter>
          <AnimatedSwitch>
            <AnimatedRoute path="/signup" fullHeight={true}>
              <span className={styles.signInHelper}>{t('auth.alreadyAnAccount')}<Link to="/signin">{t('auth.signIn.title')}</Link></span>
              <Paper className={`${styles.paper} darker`}>
                <SignUp />
              </Paper>
            </AnimatedRoute>
            <AnimatedRoute path="/signin" fullHeight={true}>
              <span className={styles.signInHelper}>{t('auth.noAccount')}<Link to="/signup">{t('auth.signUp.title')}</Link></span>
              <Paper className={`${styles.paper} darker`}>
                <SignIn />
              </Paper>
            </AnimatedRoute>
          </AnimatedSwitch>
        </BrowserRouter>
      </Container>
    </>
  );
}

