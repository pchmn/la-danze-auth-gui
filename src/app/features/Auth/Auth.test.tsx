import { Input } from '@material-ui/core';
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Auth } from './Auth';


describe('<Auth />', () => {
  const originalConsoleError = console.error;

  beforeEach(() => {
    console.error = jest.fn((msg) => {
      if (msg.includes('Warning: useLayoutEffect does nothing on the server')) {
        return null;
      } else {
        originalConsoleError(msg);
      }
    });
  });

  afterEach(() => {
    console.error = originalConsoleError;
  });

  const container = shallow(<BrowserRouter><Auth /></BrowserRouter>);
  test('it should mount', () => {
    expect(container.html()).toMatchSnapshot();
  });

  test('it should mount input', () => {
    const auth = container.find(Auth).dive();
    console.log(auth.find(Input)?.length)
    console.log(container.html(), auth)
    expect(auth?.length).toEqual(1);
  });
});