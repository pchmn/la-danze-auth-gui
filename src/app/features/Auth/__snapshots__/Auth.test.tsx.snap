// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> it should mount 1`] = `"<div class=\\"MuiContainer-root Auth  MuiContainer-maxWidthSm\\"><img class=\\"logo\\" alt=\\"logo\\" src=\\"/images/logo-48px.png\\"/></div>"`;
